src/lib.rs

```diff
@@ -133,6 +138,7 @@ impl Default for RestartConfig {
             coordination_socket_path: Default::default(),
             environment: vec![],
             lifecycle_handler: Box::new(lifecycle::NullLifecycleHandler),
+            exit_on_error: false,
```

Comment:
Should this default to `true`?  I think `true` is a safer default, because `false` means a request for a restart may be ignored and generate no error.

Reply:
Sounds good to me.